### ISSUE TITLE
requirements for current bookreader/javascript interface

### ISSUE THREAD

**Issue by verbalhanglider** [uchicago-library/digcoll_retriever#2]
- link to PDF file
- links to JPEGs
- height and width dimensions of TIFF files
- height and width dimensions of JPEG files
- custom built OCR data that includes OCR data for pages + dublin core data in header of file

**Comment by bnbalsamo** [comment]
Operating under the assumption of solution no. 2 or 1.2 from #1 I've built out some very basic endpoint tests and dummy resource classes here: https://github.com/uchicago-library/digcoll_retriever/commit/36692629bebe7a9d0cf5caf94750b0811fd1251b

Fairly basic, only question I have is more stylistic/forward thinking.

Rather than having /$identifier/jejocr would it be better to have /$identifier/ocr/jej? This allows us to later nest other potential ocr formats under the ocr endpoint as well, and in my opinion makes the interface more similar to /$identifier/jpg/stats and the like.

**Comment by verbalhanglider** [comment]
I agree about the stylistic change suggestion re /ocr/jej instead of /jejocr

my question is about stats and md? what is stats? precisely what data is going to be communicated in both endpoints? what is the purpose of stats?

**Comment by bnbalsamo** [comment]
Stats is currently where I was going to provide the dimensions of the images for the jpgs and tifs.

md is descriptive metadata, I can make that endpoint name more explanatory

**Comment by verbalhanglider** [comment]
so stats should be

 /tif/stats/dimensions
/tif/stats/dimensions/height
/tif/stats/dimensions/width

does that look right?

on that line of thinking should /md be granular too?

**Comment by verbalhanglider** [comment]
and my thinking about /md should be something like

/md/title -> title of the thing
/md/creator -> creator of the item
/md/date -> date of creation of the item
/md/publisher -> "University of Chicago Library" or "University of Chicago" or the institution responsible for this item being digitized

Of course, this all depends on having some kind of either

a. common metadata schema. this is unlikely
b. a mapper written for all metadata added to digital collections. this is the more realist option although it leaves us the job of building those mappers.

**Comment by bnbalsamo** [comment]
I'm a bit leery of splitting things up that much, and making the interface too tightly coupled to our current schemas.

I was thinking the metadata endpoint (which is now /$identifier/metadata to be more clear) might lend itself to returning either JSON or XML and being extensible to something like...

* /$identifier/metadata/dc
* /$identifier/metadata/marc
* /$identifier/metadata/$projectSpecific
* etc

That way even if in the backend we can't write a mapper for a certain project, if we have a dc record we can just throw it in and let the interface mine values.

Same idea with the stats endpoints, I think rather than going deeper down the URL chain we do something like

```
{
  "width": 800,
  "height": 600
}
```

This gives us the option to always add more stats to that JSON later, but keeps the info relatively quickly accessible in one spot to reduce the number of calls to the API required.
